packages/react-devtools-shared/src/backend/utils.js

```diff
@@ -49,12 +48,11 @@ export function copyToClipboard(value: any): void {
   // On Firefox navigator.clipboard.writeText has to be called from
   // the content script js code (because it requires the clipboardWrite
   // permission to be allowed out of a "user handling" callback),
-  // clipboardCopyText is an helper injected into the page from.
-  // injectGlobalHook.
+  // clipboardCopyText is a helper injected into the page from injectGlobalHook.
   if (typeof clipboardCopyText === 'function') {
     clipboardCopyText(text).catch(err => {});
```

Comment:
> Can you use this opportunity this verify if this is still needed? it has been a few years since this workaround was introduced #17740 #17681

Nothing changed for Firefox, as I understand. If we want to use `clipboard.writeText`, we need one of the two things:
- Asking for `"clipboardWrite"` permission. Not sure if this should be each time before calling clipboard api
- This api to be called only in user-initiated event callbacks, which will be won't work for us, since we are operating mostly as an extension

I can later test it on Firefox with this injection removed and in extension environment

Reply:
@mondaychen 

We still need this injection, because of the one use case when we are pressing "Copy value to clipboard" on some attributes of the inspected elements, like prop or field in state, or hook.

Everything from this [explanation comment](https://github.com/facebook/react/issues/17681#issuecomment-569324570) is actual.

Seems like this is the only copy operation that we do via bridge and we call clipboard api on the backend side. Since there is no user-initiated event and Firefox thinks that this is not coming from browser extension, it rejects the call.